### PR TITLE
implemented displaying user location & user location button

### PR DESCRIPTION
**Pull request by Cuiboy**
MapView now displays user location. I also added a button that jumps to user's current location.

### CODE REVIEW

File: berkeley-mobile/Map/MapViewController.swift [hunk 1]
```diff
@@ -99,14 +104,44 @@ class MapViewController: UIViewController, SearchDrawerViewDelegate {
...
+        userLocationButton.heightAnchor.constraint(equalToConstant: _buttonWidth).isActive = true
+        userLocationButton.leftAnchor.constraint(equalTo: view.layoutMarginsGuide.leftAnchor).isActive = true
+        userLocationButton.topAnchor.constraint(equalTo: filterView.bottomAnchor, constant: kViewMargin).isActive = true
+        userLocationButton.clipsToBounds = true
+        userLocationButton.layer.masksToBounds = false
+        userLocationButton.layer.cornerRadius = mapView.frame.width * 0.12 * 0.5
+        userLocationButton.layer.shadowRadius = 5
+        userLocationButton.layer.shadowOpacity = 0.2
+        userLocationButton.layer.shadowColor = UIColor.black.cgColor
+        userLocationButton.layer.shadowOffset = CGSize(width: 0, height: 10)
```
Comment (huangshawn1) [inline berkeley-mobile/Map/MapViewController.swift:132]: shadow looks a bit different from FilterViewCell but I might just be seeing wrong. want to make sure the shadow stays consistent

File: berkeley-mobile/Map/MapViewController.swift [hunk 2]
```diff
@@ -99,14 +104,44 @@ class MapViewController: UIViewController, SearchDrawerViewDelegate {
...
+        userLocationButton.setImage(UIImage(named: "Navigation")!, for: .normal)
+        userLocationButton.addTarget(self, action: #selector(jumpToUserLocation), for: .touchUpInside)
+    }
+    
+    @objc func jumpToUserLocation() {
+        let _manager = LocationManager()
+        guard let _location = _manager.userLocation else {return}
+        centerMapOnLocation(_location, mapView: mapView, animated: true)
+        userLocationButton.alpha = 0
```
Comment (huangshawn1) [inline berkeley-mobile/Map/MapViewController.swift:141]: use hidden instead

**Review by huangshawn1 — CHANGES_REQUESTED** [review]
requesting some minor changes, but looks good overall

**Review by huangshawn1 — APPROVED** [review]

**Review by Stereopoly — CHANGES_REQUESTED** [review]
Button looks great! Small nit - could you offset the position so the arrow looks a bit more centered in the circle?
Also a slight bug - If you move the map, the button disappears and never reappears. Only if I press the button, move the map, then the button reappears.

**Comment by Stereopoly** [comment]
New button works well!
Minor issue here:
Simulator Screen Shot - iPhone 11 Pro - 2020-10-03 at 14 41 41

File: berkeley-mobile/Map/MapViewController.swift [hunk 3]
```diff
@@ -87,6 +91,7 @@ class MapViewController: UIViewController, SearchDrawerViewDelegate {
         mapView.isZoomEnabled = true
         centerMapOnLocation(CLLocation(latitude: CLLocationDegrees(exactly: 37.871684)!, longitude: CLLocationDegrees(-122.259934)), mapView: mapView, animated: false)
         updateCompassPosition()
+        setUpUserLocationButton()
```
Comment (Stereopoly) [inline berkeley-mobile/Map/MapViewController.swift:94]: Move to viewDidLoad instead so it doesn't keep adding a button

**Comment by Stereopoly** [comment]
lgtm!

**Review by Stereopoly — APPROVED** [review]

**Pull request merged**